src/org/opendatakit/briefcase/export/CsvMapper.java

```diff
@@ -0,0 +1,334 @@
+  }
+
+  /**
+   * Produces a {@link List} of CSV compatible {@link String} lines with the values of
+   * a {@link Submission} instance's particular repeat group.
+   *
+   * @param groupModel      the repeat group's {@link Model}
+   * @param elements        a {@link List} of {@link XmlElement} instances to be transformed into a CSV line
+   * @param exportMedia     a {@link Boolean} indicating if we have to export media files too
+   * @param exportMediaPath the {@link Path} to the output media dir
+   * @param instanceId      the instance ID of the {@link Submission} instance that it's being exported
+   * @param workingDir      the {@link Path} to the working directory of the {@link Submission} instance that it's being exported
+   * @return a {@link List} of CSV compatible {@link String} lines
+   */
+  static String getRepeatCsvLine(Model groupModel, List<XmlElement> elements, boolean exportMedia, Path exportMediaPath, String instanceId, Path workingDir) {
+    StringBuilder sb = new StringBuilder();
+    elements.forEach(element -> {
+      String localId = element.getCurrentLocalId(instanceId);
+      StringBuilder sb2 = new StringBuilder();
+      groupModel.forEach(field -> getMapper(field).apply(
+          localId,
+          workingDir,
+          field,
+          element.findElement(field.getName()),
+          exportMedia,
+          exportMediaPath
+      ).forEach(pair -> sb2.append(",").append(encode(pair.right, pair.left.startsWith("meta") || pair.left.startsWith("SET-OF")))));
+
+      sb2.append(",").append(encode(element.getParentLocalId(instanceId), false));
+      sb2.append(",").append(encode(element.getCurrentLocalId(instanceId), false));
+      sb2.append(",").append(encode(element.getGroupLocalId(instanceId), false));
+      sb.append(sb2.toString().substring(1)).append("\n");
+    });
+    return sb.toString();
+  }
+
+  private static FieldMapper getMapper(Model field) {
+    return Optional.ofNullable(mappers.get(field.getDataType()))
+        // If no mapper has been defined, we'll just output the text
+        .orElse(simpleMapper(CsvMapper::text));
+  }
+
+  private static List<Pair<String, String>> text(XmlElement element) {
+    return singletonList(Pair.of(
+        element.fqn(),
+        element.maybeValue().orElse(null)
+    ));
+  }
+
+  private static List<Pair<String, String>> date(XmlElement element) {
+    return formattedDate(element, getDateInstance());
+  }
+
+  private static List<Pair<String, String>> time(XmlElement element) {
+    return formattedDate(element, getTimeInstance());
+  }
+
+  private static List<Pair<String, String>> dateTime(XmlElement element) {
+    return formattedDate(element, getDateTimeInstance());
+  }
+
+  private static List<Pair<String, String>> geopoint(XmlElement element) {
+    String[] fields = Optional.ofNullable(element.getValue())
+        .map(v -> v.split(" "))
+        .orElseThrow(() -> new BriefcaseException("No value on geopoint field"));
```

Comment:
This also throws an exception if element is empty, presumably not completing the export.